internal/confmapprovider/discovery/discoverer.go

```diff
@@ -117,6 +124,34 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 	return d, nil
 }
 
+func (d *discoverer) resolveConfig(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
+	providers := map[string]confmap.Provider{
+		"yaml": yamlProvider,
+		"env":  envProvider,
+	}
+	out, err := yaml.Marshal(discoveryReceiverRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal discovery receiver config for uri: %w", err)
+	}
+	uris := []string{fmt.Sprintf("yaml:%s", out)}
+	resolver, err := confmap.NewResolver(confmap.ResolverSettings{
+		URIs:       uris,
+		Providers:  providers,
+		Converters: []confmap.Converter{d.expandConverter},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a resolver from the given uris. %w", err)
+	}
+
+	conf, err := resolver.Resolve(context.Background())
+	if err != nil {
+		d.logger.Error(fmt.Sprintf("failed to resolve configuration as a conf from the resovler %v", err))
+		return nil, fmt.Errorf("failed to resolve configuration as a conf from the resovler %w", err)
```

Comment:
think a commit was lost

```suggestion
		return nil, fmt.Errorf("failed to resolve configuration from the resolver %w", err)
```